web/packages/teleterm/src/ui/services/workspacesService/documentsService/documentsService.ts

```diff
@@ -352,13 +352,18 @@ export class DocumentsService {
    */
   update(
     uri: DocumentUri,
-    updated: Partial<Document> | ((document: Document) => Document)
+    updated: Partial<Document> | ((document: Document) => void | Document)
   ) {
     this.setState(draft => {
       const index = draft.documents.findIndex(doc => doc.uri === uri);
       const toUpdate = draft.documents[index];
       if (typeof updated === 'function') {
-        draft.documents[index] = updated(toUpdate);
+        const newState = updated(toUpdate);
+        // if a new object was returned,
+        // we have to set it at the appropriate index
+        if (newState) {
+          draft.documents[index] = newState;
```

Comment:
Ah cool, I didn't know about returning new state in Immer. Could we leave a comment like "Support returning new state https://immerjs.github.io/immer/return/"?